### PR TITLE
Feature/cu p3d27v melee grunt rework

### PR DESCRIPTION
**Pull request by Yonimevicio**
Now meleegrunt has 3 attacks
Attackspeed is linked to de animation speed and it can be changed
All animations should work correctly
Now RightBlade and LeftBlade are searched in the script by name (their names can't be changed)
There are a lot of new references in the script references (attackspeed, movementspeed, chance of attack1,2 ,etc.)
TestScene: MeleeGruntRework
Prefab: MeleeGruntReworkedV1_Level1.prefab

### CODE REVIEW

**Comment by jcarlos0305** [comment]
Task linked: [CU-p3d27v Melee Grunt - Add new attacks / deaths](https://app.clickup.com/t/p3d27v)

File: Penteract/Assets/Scripts/AIMeleeGrunt.cpp [hunk 1]
```diff
@@ -186,33 +194,44 @@ void AIMeleeGrunt::Update() {
...
-		attackRemaining -= Time::GetDeltaTime();
-		if (attackRemaining < 1.3f) { //frame 160
-			movementScript->Stop(); // stop seek
+		if (track) { 
+			movementScript->Orientate(player->GetComponent<ComponentTransform>()->GetGlobalPosition() - ownerTransform->GetGlobalPosition()); 
 		}
-		else {
-			movementScript->Orientate(player->GetComponent<ComponentTransform>()->GetGlobalPosition() - ownerTransform->GetGlobalPosition());
+		if (attackStep) {
+			movementScript->Seek(state, ownerTransform->GetGlobalPosition() + ownerTransform->GetFront()*10, attackMovementSpeed, false);
```
Comment (magalenyo) [inline Penteract/Assets/Scripts/AIMeleeGrunt.cpp:225]: Maybe expose the *10 so the distance multiplier can be tweaked?

**Review by magalenyo — APPROVED** [review]

File: Penteract/Assets/Scripts/AIMeleeGrunt.cpp (hunk 1 above)
Comment (Yonimevicio) [reply Penteract/Assets/Scripts/AIMeleeGrunt.cpp:225]: It is only to make sure u go front It would not have to change.

**Review by jcarlos0305 — DISMISSED** [review]

File: Penteract/Assets/Scripts/AIMovement.cpp [hunk 2]
```diff
@@ -95,3 +96,27 @@ bool AIMovement::CharacterInAttackRange(const GameObject* character, const float
 
 	return false;
 }
+
+void AIMovement::SetClipSpeed(UID clipUID, float speed) {
+	if (speed >= 0.f) {
+		ResourceClip* clip = GameplaySystems::GetResource<ResourceClip>(clipUID);
+		clip->speed = speed;
+	}
+}
```
Comment (Mefiso) [inline Penteract/Assets/Scripts/AIMovement.cpp:105]: Why is this here? It seems either more generic than "movement" or specific to the Melee grunt (if it will only be used by him). So I don't know if it should be here in AIMovement

File: Penteract/Assets/Scripts/AIMovement.cpp [hunk 3]
```diff
@@ -95,3 +96,27 @@ bool AIMovement::CharacterInAttackRange(const GameObject* character, const float
...
+	}
+	GameObject* reference = nullptr;
+	for (GameObject* child : root->GetChildren())
+	{
+		reference = SearchRefenceInHierarchy(child, name);
+		if (reference != nullptr) return reference;
+
+	}
+	return nullptr;
+}
```
Comment (Mefiso) [inline Penteract/Assets/Scripts/AIMovement.cpp:120]: Same

File: Penteract/Assets/Scripts/AIMeleeGrunt.cpp [hunk 4]
```diff
@@ -87,7 +98,10 @@ void AIMeleeGrunt::Start() {
 	}
 
 	movementScript = GET_SCRIPT(&GetOwner(), AIMovement);
-
+	rightBladeCollider = movementScript->SearchRefenceInHierarchy(&GetOwner(),"RightBlade");
```
Comment (Mefiso) [inline Penteract/Assets/Scripts/AIMeleeGrunt.cpp:101]: By Mephistopheles, RefeREnce not Refence, pls.

**Review by Mefiso — CHANGES_REQUESTED** [review]
Please at least change the name of the method SearchRefenceInHierarchy.

File: Penteract/Assets/Scripts/AIMovement.cpp (hunk 2 above)
Comment (Yonimevicio) [reply Penteract/Assets/Scripts/AIMovement.cpp:105]: I don't have another generic script for this kind of functions and i think we can change the name of the script because there are a lot of other no-movement functions there.

File: Penteract/Assets/Scripts/AIMeleeGrunt.cpp (hunk 4 above)
Comment (Yonimevicio) [reply Penteract/Assets/Scripts/AIMeleeGrunt.cpp:101]: My fault :).

**Review by Mefiso — APPROVED** [review]

**Review by magalenyo — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]
